skbeam/core/correlation.py

```diff
@@ -405,3 +409,356 @@ def auto_corr_scat_factor(lags, beta, relaxation_rate, baseline=1):
+    The two-time correlation function is defined as
+
+    :math ::
+        C(q, t_1, t_2) = \frac{<I(q, t_1)I(q, t_2)>_pix }{<I(q, t_1)>_pix <I(q, t_2)>_pix}
+
+    Here, the ensemble averages are performed over many pixels of detector,
+    all having the same q value. The average time or age is equal to (t1+t2)/2,
+    measured by the distance along the t1 = t2 diagonal.
+    The time difference t = |t1 - t2|, with is distance from the t1 = t2
+    diagonal in the perpendicular direction.
+    In the equilibrium system, the two-time correlation functions depend only
+    on the time difference t, and hence the two-time correlation contour lines
+    are parallel.
+
+    References
+    ----------
+
+    .. [1] A. Fluerasu, A. Moussaid, A. Mandsen and A. Schofield,
+        "Slow dynamics and aging in collodial gels studied by x-ray photon
+         correlation spectroscopy," Phys. Rev. E., vol 76, p 010401(1-4), 2007.
+    """
+    if two_time_internal_state is None:
+        two_time_internal_state = _init_state_two_time(num_levels, num_bufs,
+                                                       labels, num_frames)
+    # create a shorthand reference to the results and state named tuple
+    s = two_time_internal_state
+
+    for img in images:
+        s.cur[0] = (1 + s.cur[0]) % num_bufs  # increment buffer
+
+        s.count_level[0] = 1 + s.count_level[0]
+
+        # get the current image time
+        s = s._replace(current_img_time=(s.current_img_time + 1))
+
+        # Put the image into the ring buffer.
+        s.buf[0, s.cur[0] - 1] = (np.ravel(img))[s.pixel_list]
+
+        # Compute the two time correlations between the first level
+        # (undownsampled) frames. two_time and img_per_level in place!
+        _two_time_process(s.buf, s.two_time, s.label_array, num_bufs,
+                          s.num_pixels, s.img_per_level, s.lag_steps,
+                          s.current_img_time, level=0, buf_no=s.cur[0] - 1)
+
+        # time frame for each level
+        s.time_ind[0].append(s.current_img_time)
+
+        # check whether the number of levels is one, otherwise
+        # continue processing the next level
+        processing = num_levels > 1
+
+        # Compute the correlations for all higher levels.
+        level = 1
+        while processing:
+            if not s.track_level[level]:
+                s.track_level[level] = 1
+                processing = False
+            else:
+                prev = 1 + (s.cur[level - 1] - 2) % num_bufs
+                s.cur[level] = 1 + s.cur[level] % num_bufs
+                s.count_level[level] = 1 + s.count_level[level]
+
+                s.buf[level, s.cur[level] - 1] = (s.buf[level - 1, prev - 1] +
+                                                  s.buf[level - 1,
+                                                  s.cur[level - 1] - 1])/2
+
+                t1_idx = (s.count_level[level] - 1) * 2
+
+                current_img_time = ((s.time_ind[level - 1])[t1_idx]
```

Comment:
Yes, you don't have to track it, this will only use in multi_tau method, current_img_time will be calculated here, it doesn't need prior values for it.